Source/WebKit/Configurations/_WebKit_SwiftUI.xcconfig

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2024 Apple Inc. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "BaseTarget.xcconfig"
+
+// InstallAPI is transitively enabled through including BaseTarget.xcconfig.
+
+PRODUCT_NAME = _WebKit_SwiftUI; // Cross-import overlay projects must have this exact naming scheme.
+PRODUCT_BUNDLE_IDENTIFIER = com.apple.WebKit.$(PRODUCT_NAME:rfc1034identifier);
+INFOPLIST_FILE = Info.plist;
+
+INSTALL_PATH = $(WEBKIT_FRAMEWORKS_DIR);
+DYLIB_INSTALL_NAME_BASE = $(NORMAL_WEBKIT_FRAMEWORKS_DIR);
+
+ALTERNATE_ROOT_PATH = $(NORMAL_WEBKIT_FRAMEWORKS_DIR);
+
+OUTPUT_ALTERNATE_ROOT_PATH = $(OUTPUT_ALTERNATE_ROOT_PATH_$(USE_SYSTEM_CONTENT_PATH)_$(WK_NOT_$(WK_USE_OVERRIDE_FRAMEWORKS_DIR)));
+OUTPUT_ALTERNATE_ROOT_PATH_YES_YES = $(DSTROOT)$(ALTERNATE_ROOT_PATH)/$(FULL_PRODUCT_NAME);
+
+BUILD_LIBRARY_FOR_DISTRIBUTION = YES; // This is required for all public frameworks in the OS.
+
+SWIFT_LIBRARY_LEVEL = api;
+
+// Prevent Objective-C from importing the cross-import overlay
+SWIFT_INSTALL_OBJC_HEADER = NO;
+DEFINES_MODULE = NO;
+
+OTHER_SWIFT_FLAGS = -Xfrontend -experimental-spi-only-imports $(OTHER_SWIFT_FLAGS_AVAILABILITY_$(USE_INTERNAL_SDK));
```

Comment:
We set this in Base.xcconfig, so you shouldn't need it here.

Reply:
I initially had that, but realized I don't want the `-no-warnings-as-errors` setting since I want all warnings to be errors in this project

Reply:
Ah, thanks for explaining your intention here.

I don't think that's feasible, because Swift doesn't provide a way to temporarily silence a deprecation or syntax warning (rdar://26611918). That's essential for our project, because we build for multiple OS releases and sometimes have to retain old syntax that has become a warning in recent compilers (e.g. `@_implementationOnly`), and because we need to rapidly respond to deprecations to avoid causing internal build breakage.

Maybe you can get away with warnings as errors today, since this is a new feature, but you won't be able to long term. And regardless, I think that these policies should be set project-wide whenever possible.